src/test/scala/vsys/api/http/ApiErrorSpecification.scala

```diff
@@ -0,0 +1,246 @@
+    result.message should be("invalid database datatype SuperNode")
+  }
+
+  property("WrongFeeScale should cause err which id is 114") {
+    forAll(feeScaleGen) { feeScale => 
+      val err = ValidationError.WrongFeeScale(feeScale) 
+      val result = ApiError.fromValidationError(err)
+      result.id should be(ToSelfError.id)
+      result.code should be(StatusCodes.BadRequest)
+      result.message should be(s"Validation failed for wrong fee scale $feeScale.")
+    }
+  }
+
+  property("InvalidSlotId should cause err which id is 116") {
+    forAll (slotidGen) { slotId => 
+      val err = ValidationError.InvalidSlotId(slotId) 
+      val result = ApiError.fromValidationError(err)
+      result.id should be(InvalidSlotId(slotId).id)
+      result.code should be(StatusCodes.BadRequest)
+      result.message should be(s"slot id: $slotId invalid.")
+    }
+  }
+
+  property("InvalidMintingReward") {
+    forAll(mintingAmountGen){ errMintingReward => 
+      val err = ValidationError.WrongMintingReward(errMintingReward)
+      val result = ApiError.fromValidationError(err)
+      result.id should be(InvalidMintingReward(errMintingReward).id)
+      result.code should be(StatusCodes.BadRequest)
+      result.message should be(s"Validation failed for wrong minting reward $errMintingReward.")
+    }
+  }
+  
+  property("TooLongDbEntry should throw error if actual length is greater than maxLength") {
+    val maxLength = Entry.maxLength
+    val actualLength = maxLength + 2
+
+    val err = ValidationError.TooLongDbEntry(actualLength, maxLength) 
+    val result = ApiError.fromValidationError(err)
+    result.id should be(TooLongDbEntry(actualLength, maxLength).id)
+    result.code should be(StatusCodes.BadRequest)
+    result.message should be(s"The data is too long for database put, the actual length " +
+      s"is $actualLength, the maximun length supported for now is $maxLength")
+    
+  }
+  property("InvalidUTF8String") {
+    val err = ValidationError.InvalidUTF8String("contractDescription") 
+    val result = ApiError.fromValidationError(err)
+    result.id should be(InvalidUTF8String("contractDescription").id)
+    result.code should be(StatusCodes.BadRequest)
+    result.message should be(s"The contractDescription is not a valid utf8 string")
+  }
+
+  property("TransactionValidationError should return error according to error type TransactionValidationError contains") {
+    forAll (randomTransactionGen) { transaction =>
+      val err = ValidationError.Mistiming("Mistiming error")
+      val invalidTransacErr = TransactionValidationError(err, transaction)
+      val result = ApiError.fromValidationError(invalidTransacErr)
+      result.id should be(Mistiming.Id)
+      result.code should be(StatusCodes.BadRequest)
+      result.message should be("Mistiming error")
+    }
+  }
+
+  property("Unknown should be error which has specific id, code and message") {
+    val result = Unknown
+    result.id should be(Unknown.id)
```

Comment:
@zl730 @Young666666Jian does it necessary to check the id here? since u set the `result = Unkown`